src/main/java/com/utilaider/logistics/controller/LogisticsController.java

```diff
@@ -142,21 +188,15 @@ public ModelAndView completeProfile(@ModelAttribute Owner owner, BindingResult r
                 fetchedOwner.setNoOfVehicles(owner.getNoOfVehicles());
                 fetchedOwner.setBusinessType(owner.getBusinessType());
                 fetchedOwner.setUserReferanceCode(owner.getUserReferanceCode());
-                owner.getAddress().setPincode(Integer.parseInt(request.getParameter("pincode111")));
+                owner.getAddress().setUser(fetchedOwner);
```

Comment:
Looks likec cyclic reference that might be uncessary.

owner has an address
An address belongs to an owner